### PR TITLE
ADR and implementation for client-side feature flags

### PR DESCRIPTION
**Pull request by jim**
## Description

This PR adds a simple way to toggle features on and off in the React app. Please read the ADR for more background on what went into this approach and ask all the questions you can think of.

## Reviewer Notes

During a conversation with @tinyels, we decided that we should use `AppContext` to provide the feature flags throughout the app.

Here is an example of how to access the feature flags from within React:

```jsx
<AppContext.Consumer>
  {settings => (
    <p>HHG is {settings.flags.hhg ? 'enabled' : 'disabled'}.</p>
  )}
</AppContext.Consumer>
```

You can additionally override any flag setting using query params. This is intended for use mostly by e2e tests, but it might some in handy for other purposes.

To override the value of the `hhg` and flip it off, you can access the app and append `?flag:hhg=false` to the URL during the initial app load. Note that these query string values are only read when the app initially boots.

## Code Review Verification Steps

* [ ] All tests pass.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/158741324) for this change

### CODE REVIEW

**Comment by tinyels** [comment]
A couple of questions/concerns about using hostnames for this:

1. I don't run my local dev environment as localhost. I use a set of environment variables so that I can access it as "mymove" on my windows vm. How will that play out with using hostnames for this?
2. Ideally we will be building e2e tests for new features. The new e2e make task runs those against a "production" build of the client code. Will it be possible to test new features? Will it be possible to write a test that ensures feature is covered by feature flag (e.g. set flag for test)?

File: src/shared/featureFlags.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  // If we've built the app, then use the hostname to determine what the
+  // environment is.
+  const domain = window.location.host;
+  switch (domain) {
+    case 'office.move.mil':
+    case 'my.move.mil':
+      return 'production';
+      break;
+    case 'office-staging.move.mil':
+    case 'my-staging.move.mil':
```
[the remark this answers is not in the document]
Comment (akostibas) [reply src/shared/featureFlags.js]: Perhaps move these to the top as constants, so they're easier to find / update?

[the remark this answers is not in the document]
Comment (jim) [reply src/shared/featureFlags.js]: I can go either way on adding top-level variables. They're only used in one place, so I am inclined to not add them at this point. If (or anyone else) feels strongly, I will add them.

**Comment by jim** [comment]
@akostibas Ready for another look!

File: src/scenes/Moves/MoveType.jsx [hunk 2]
```diff
@@ -193,6 +194,11 @@ export class MoveType extends Component {
           selectedOption={currentOption}
           onMoveTypeSelected={this.onMoveTypeSelected}
         />
+        <AppContext.Consumer>
+          {settings => (
+            <p>HHG is {settings.flags.hhg ? 'enabled' : 'disabled'}.</p>
```
[the remark this answers is not in the document]
Comment (jim) [reply src/scenes/Moves/MoveType.jsx]: Done!

**Comment by jim** [comment]
@akostibas Should be ready now.

**Pull request merged**